src/cmd/initialize.go

```diff
@@ -44,46 +50,55 @@ var initCmd = &cobra.Command{
 
-		err := validateInitFlags()
-		if err != nil {
+		if err := validateInitFlags(); err != nil {
 			message.Fatal(err, "Invalid command flags were provided.")
 		}
 
 		// Continue running package deploy for all components like any other package
-		initPackageName := config.GetInitPackageName()
-		config.DeployOptions.PackagePath = initPackageName
+		initPackageName := packager.GetInitPackageName("")
+		pkgConfig.DeployOpts.PackagePath = initPackageName
 
 		// Try to use an init-package in the executable directory if none exist in current working directory
-		if utils.InvalidPath(config.DeployOptions.PackagePath) {
-			// Get the path to the executable
-			if executablePath, err := utils.GetFinalExecutablePath(); err != nil {
-				message.Errorf(err, "Unable to get the path to the executable")
-			} else {
-				executableDir := path.Dir(executablePath)
-				config.DeployOptions.PackagePath = filepath.Join(executableDir, initPackageName)
-			}
-
-			// If the init-package doesn't exist in the executable directory, try the cache directory
-			if err != nil || utils.InvalidPath(config.DeployOptions.PackagePath) {
-				config.DeployOptions.PackagePath = filepath.Join(config.GetAbsCachePath(), initPackageName)
-
-				// If the init-package doesn't exist in the cache directory, suggest downloading it
-				if utils.InvalidPath(config.DeployOptions.PackagePath) {
-					if err := downloadInitPackage(initPackageName); err != nil {
-						message.Fatal(err, "Failed to download the init package")
-					}
-				}
-			}
+		var err error
+		if pkgConfig.DeployOpts.PackagePath, err = findInitPackage(initPackageName); err != nil {
+			message.Fatal(err, err.Error())
 		}
 
 		// Run everything
-		packager.Deploy()
+		packager.NewOrDie(&pkgConfig).Deploy()
 	},
 }
 
+func findInitPackage(initPackageName string) (string, error) {
+	// First, look for the init package in the current working directory
+	if !utils.InvalidPath(initPackageName) {
+		return initPackageName, nil
+	}
+
+	// Next, look for the init package in the executable directory
+	executablePath, err := utils.GetFinalExecutablePath()
+	if err != nil {
+		return "", err
+	}
+	executableDir := path.Dir(executablePath)
+	if !utils.InvalidPath(filepath.Join(executableDir, initPackageName)) {
+		return filepath.Join(executableDir, initPackageName), nil
+	}
+
+	// Finally, if the init-package doesn't exist in the cache directory, suggest downloading it
+	if err := downloadInitPackage(initPackageName); err != nil {
+		if errors.Is(err, lang.ErrInitNotFound) {
+			message.Fatal(err, err.Error())
+		} else {
+			message.Fatalf(err, "Failed to download the init package: %w", err)
+		}
+	}
+	return "", nil
+}
+
 func downloadInitPackage(initPackageName string) error {
 	if config.CommonOptions.Confirm {
-		return fmt.Errorf("this command requires a zarf-init package, but one was not found on the local system")
+		return lang.ErrInitNotFound
 	}
 
 	var confirmDownload bool
```

Comment:
As discussed (in person-ish), this PR will set the baseline for lang file capability and we'll slowly go about utilizing the lang file as we complete future work instead of also trying to cram that into this PR.